pkg/pod/creds_init.go

```diff
@@ -63,6 +68,11 @@ func credsInit(serviceAccountName, namespace string, kubeclient kubernetes.Inter
 			return nil, nil, nil, err
 		}
 
+		err = checkGitSSHSecret(ctx, secret)
```

Comment:
Go style nit: If we can, generally we try to write cases like this as:

```go
if err := checkGitSSHSecret(ctx, secret); err != nil {
  return nil, nil, nil, err
}
```

This keeps the returned error variable in the smallest possible scope, and doesn't change the value of the previous `err` value.

Over time this form becomes easy to read at a glance as, "try this; if it returns an error, return it"